### PR TITLE
Generate docs with TBDocs

### PR DESCRIPTION
**Pull request by leordev**
@frankhinek please let me know if you are happy with this @web5/api docs generation automation.

Preview image:
image

Artifact to be downloaded and tested locally: https://github.com/TBD54566975/web5-js/actions/runs/7215382638

### CODE REVIEW

File: .github/workflows/docs-publish.yml [hunk 1]
```diff
@@ -2,10 +2,11 @@
 name: Build and Deploy Docs
 
 on:
-  # Runs on pushes targeting the default branch
-  # push:
-  #   branches:
-  #     - main
+  # TODO: uncomment before merging
```
Comment (leordev) [inline .github/workflows/docs-publish.yml:5]: automatic docs publishing when a new release is cut

**Comment by angiejones** [comment]
what do you all think about including all of the web5 packages here, namely dids, credentials, and crypto? Or should they have their own docs?

**Comment by leordev** [comment]
> what do you all think about including all of the web5 packages here, namely dids, credentials, and crypto? Or should they have their own docs?

We are planning exactly on that, but first we need to fix the doc errors!

**Comment by leordev** [comment]
Closing in favor of #402

**Pull request closed without merging**